cluster/gce/gci/configure-helper.sh

```diff
@@ -1942,21 +1942,47 @@ function copy-manifests {
 # Fluentd manifest is modified using kubectl, which may not be available at
 # this point. Run this as a background process.
 function wait-for-apiserver-and-update-fluentd {
+  local -r fluentd_gcp_yaml="${1}"
+
+  local modifying_flags=""
+  if [[ -n "${FLUENTD_GCP_MEMORY_LIMIT:-}" ]]; then
+    modifying_flags="${modifying_flags} --limits=memory=${FLUENTD_GCP_MEMORY_LIMIT}"
+  fi
+  local request_resources=""
+  if [[ -n "${FLUENTD_GCP_CPU_REQUEST:-}" ]]; then
+    request_resources="cpu=${FLUENTD_GCP_CPU_REQUEST}"
+  fi
+  if [[ -n "${FLUENTD_GCP_MEMORY_REQUEST:-}" ]]; then
+    if [[ -n "${request_resources}" ]]; then
+      request_resources="${request_resources},"
+    fi
+    request_resources="memory=${FLUENTD_GCP_MEMORY_REQUEST}"
+  fi
+  if [[ -n "${request_resources}" ]]; then
+    modifying_flags="${modifying_flags} --requests=${request_resources}"
+  fi
+
   until kubectl get nodes
   do
     sleep 10
   done
-  kubectl set resources --dry-run --local -f ${fluentd_gcp_yaml} \
-    --limits=memory=${FLUENTD_GCP_MEMORY_LIMIT} \
-    --requests=cpu=${FLUENTD_GCP_CPU_REQUEST},memory=${FLUENTD_GCP_MEMORY_REQUEST} \
-    --containers=fluentd-gcp -o yaml > ${fluentd_gcp_yaml}.tmp
-  mv ${fluentd_gcp_yaml}.tmp ${fluentd_gcp_yaml}
+
+  local -r temp_fluentd_gcp_yaml="${fluentd_gcp_yaml}.tmp"
+  if kubectl set resources --dry-run --local -f ${fluentd_gcp_yaml} ${modifying_flags} \
+      --containers=fluentd-gcp -o yaml > ${temp_fluentd_gcp_yaml}; then
+    mv ${temp_fluentd_gcp_yaml} ${fluentd_gcp_yaml}
+  else
+    echo "Failed to update fluentd resources"
```

Comment:
>&2 to use stderr instead of stdout? Also, if the yaml file is going to be deleted, maybe just print its contents as well? It will be easier to figure out what went wrong if this fails.

Reply:
SG, thanks